job_executor/model/metadata.py

```diff
@@ -295,7 +256,7 @@ def patch(self, other: "Variable") -> "Variable":
                 patched_represented_variables.append(
                     self.represented_variables[idx]
                     .patch(other.represented_variables[idx])
-                    .dict()
+                    .model_dump()
```

Comment:
Let's use the aliases, and exclude nones here too for the sake of consistency 🙏🏻